### PR TITLE
fix pks-4072 windows drain cluster

### PR DESCRIPTION
**Pull request by hongtaiwang**
add drain-cluster job for windows

### CODE REVIEW

**Comment by ahrtr** [comment]
Why do you add an empty file " jobs/drain-cluster-windows/monit "?

File: jobs/drain-cluster-windows/spec [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+name: drain-cluster-windows
+
+templates:
+  bin/run.ps1: bin/run.ps1
```
Comment (ahrtr) [inline jobs/drain-cluster-windows/spec:5]: When will the "run.ps1" be executed, in other words, who will trigger this script?  Do you have a separate PR for this task?

Comment (hongtaiwang) [reply jobs/drain-cluster-windows/spec:5]: yes, i added drain-cluster-windows as a pre-delete errand in p-pks-integrations. It will be executed in the same stage as drain-cluster errand for linux clusters
will submit the integration pr together with the main build for service-adapter and kubo-windows

**Comment by hongtaiwang** [comment]
> Why do you add an empty file " jobs/drain-cluster-windows/monit "?

it is generated by  `bosh generate-job`  . I think it's part of bosh job template. We don't need monit process for this job, thus left it empty

File: jobs/drain-cluster-windows/spec (hunk 1 above)
Comment (YanzhaoLi) [reply jobs/drain-cluster-windows/spec:5]: Why do you choose pre-stop instead of drain

Comment (hongtaiwang) [reply jobs/drain-cluster-windows/spec:5]: correct, it will run only when deleting cluster. I don't think i should change the behavior for other commands

Comment (YanzhaoLi) [reply jobs/drain-cluster-windows/spec:5]: For linux platform, will `kubectl drain` happen when upgrading cluster?

Comment (hongtaiwang) [reply jobs/drain-cluster-windows/spec:5]: for linux, the kubelet drain will run in pre-stop stage when upgrading cluster.

**Comment by ahrtr** [comment]
@hongtaiwang I suggest you to provide a design document for all the reviewers to review firstly. It should have the high level design/ideas and all the detailed implementations on all related repositories.

**Comment by hongtaiwang** [comment]
> @hongtaiwang I suggest you to provide a design document for all the reviewers to review firstly. It should have the high level design/ideas and all the detailed implementations on all related repositories.

https://confluence.eng.vmware.com/display/~whongtai/pks-4072+Failed+to+delete+windows+cluster+with+PV+attached

**Comment by ahrtr** [comment]
Have you performed e2e test? If yes, please list the test steps. 

Please also make sure a RaaS pipeline can pass.

**Comment by hongtaiwang** [comment]
> Have you performed e2e test? If yes, please list the test steps.
> 
> Please also make sure a RaaS pipeline can pass.

yes, since no field to add pipeline info in PR for this repo, I`ve put them on the service-adapter PR.
> pks-api pipeline:
> https://pks.ci.cf-app.com/teams/bosh-lifecycle-dev/pipelines/pks-api-1.11.x-wht-test-windowsdraincluster
> raas pipeline:
> https://raas.gcp.pks-releng.cf-app.com/teams/nsx-t/pipelines/vsphere70-nsx31-om210-install-wht-fix-windows-drain-cluster-wht-fix-windows-drain-cluster2
> https://raas.gcp.pks-releng.cf-app.com/teams/nsx-t/pipelines/vsphere70-nsx31-om210-upgrade-minor-wht-fix-windows-drain-cluster-wht-fix-windows-drain-cluster2
Just paste the pipeline links in the doc as well.

**Review by ahrtr — APPROVED** [review]
